types/Yargs.ts

```diff
@@ -35,3 +35,16 @@ export type ProjectDevArgs = CommonArgs & ConfigArgs & EnvironmentArgs;
 export type TestingArgs = {
   qa?: boolean;
 };
+
+// This is a workaround to make the builder method required for the CommandModule
+export type YargsCommandModule<T, U> = Omit<CommandModule<T, U>, 'builder'> & {
+  builder: (yargs: Argv) => Promise<Argv<U>>;
+};
```

Comment:
Nit: Since `CommandModule` is an interface we can extend it and it might simplify the TS logic here.

```typescript
export interface CommandModuleWithRequiredBuilder<T, U> extends CommandModule<T, U> {
  builder: (yargs: Argv) => Promise<Argv<U>>
}
```